File: C++/DFS-BFS.cpp
/* 
G = Graph, V = Vertices/Nodes, E = Edges/Links
G = (V, E)
*/
#include<iostream>
#include<vector>
#include<queue>

using namespace std;

void DFS(int V, vector<vector<int>> &G, vector<bool> &visited){

    visited[V] = true;
    cout << V << endl;

    vector<int> adj = G[V];

    for(int i = 0; i < adj.size(); i++){
        if(!visited[adj[i]])
            DFS(adj[i], G, visited);
    }

}

void BFS(int V, vector<vector<int>> &G, vector<bool> &visited){

    int currentV;
    vector<int> adj;

    visited[V] = true;

    queue<int> Q;
    Q.push(V);

    while(!Q.empty()){

        currentV = Q.front();
        cout << currentV << endl;
        Q.pop();
        adj = G[currentV];

        for(int i = 0; i < adj.size(); i++){

            if(!visited[adj[i]]){
                visited[adj[i]] = true;
                Q.push(adj[i]);
            }

        }

    }

}


int main(){

    vector<vector<int>> G(7);

    G[0].push_back(4);
    G[0].push_back(5);
    G[0].push_back(1);
    
    G[1].push_back(0);
    G[1].push_back(2);
    G[1].push_back(3);
    G[1].push_back(4);
    
    G[2].push_back(1);
    G[2].push_back(5);
    
    G[3].push_back(1);
    
    G[4].push_back(0);
    G[4].push_back(1);
    
    G[5].push_back(0);
    G[5].push_back(2);
    
    vector<bool> visited(7, false);

    cout << "DFS" << endl;
    DFS(0, G, visited);

    visited = vector<bool>(7, false);

    cout << "BFS" << endl;
    BFS(0, G, visited);

    return 0;
}

File: Python/DFS-BFS.py
from collections import deque

def DFS(currentNode):
    visited[currentNode] = True
    print(currentNode)
    adjacent = graph[currentNode]
    for next in adjacent:
        if(not visited[next]):
            DFS(next)

def BFS(start):
    q = deque()
    q.append(start)
    visited[start] = True
    while(q):
        current = q[0]
        q.popleft()
        print(current)
        for adj in graph[current]:
            if(not visited[adj]):
                q.append(adj)
                visited[adj] = True

def main():
    global graph, visited
    graph = []
    graph.append([4,5,1])
    graph.append([0,2,3,4])
    graph.append([1,5])
    graph.append([1])
    graph.append([0,1])
    graph.append([0,2])
    graph.append([])
    visited = [False for _ in graph]
    print(graph)
    print(visited)
    print("DFS")
    DFS(0)
    visited = [False for _ in graph]
    print("BFS")
    BFS(0)
main()